src/vs/code/electron-main/app.ts

```diff
@@ -1471,4 +1482,49 @@ export class CodeApplication extends Disposable {
 		// Refs: https://github.com/microsoft/vscode/issues/234064
 		validatedevDeviceId(this.stateService, this.logService);
 	}
+
+	private async moveToApplicationsFolder(accessor: ServicesAccessor): Promise<void> {
+		const dialogMainService = accessor.get(IDialogMainService);
+
+		const { response } = await dialogMainService.showMessageBox({
+			type: 'info',
+			buttons: [
+				localize({ key: 'move', comment: ['&& denotes a mnemonic'] }, "&&Move to Applications folder"),
+				localize({ key: 'doNotMove', comment: ['&& denotes a mnemonic'] }, "&&Cancel")
+			],
+			message: localize('moveToApplicationsFolderWarning', "Current version of {0} is installed outside the Applications folder, would you like to move it ?", this.productService.nameLong),
+			checkboxLabel: localize('remember', "Do not ask again"),
+			cancelId: 1
+		});
+
+		if (response === 0) {
+			try {
+				const result = app.moveToApplicationsFolder({
+					conflictHandler: conflictType => {
+						if (conflictType === 'exists') {
+							const response = dialog.showMessageBoxSync(massageMessageBoxOptions({
+								type: 'warning',
+								buttons: [
+									localize({ key: 'continue', comment: ['&& denotes a mnemonic'] }, "&&Yes"),
+									localize({ key: 'cancel', comment: ['&& denotes a mnemonic'] }, "&&No")
+								],
+								message: localize('applicationAlreadyExists', "A version of {0} already exists inside the Applications folder, would you like to replace it ?", this.productService.nameLong),
+								detail: localize('applicationAlreadyExistsDetail', "Continuing this step would proceed to replace the version inside the Applications folder"),
```

Comment:
This reads very verbose, can we tighten it up?

Reply:
@isidorn suggested some alternate wording, will update the PR.